### DISCUSSION TITLE
OpenTelemetry logs / metrics semantics

### DISCUSSION THREAD

**Discussion by deepkumar-git** [open-telemetry/community#981] [Community Discussions]
Hi,
I am new to OpenTelemetry and is still learning. We are thinking of using Otel as an interoperable vendor agnostic layer for standardizing logs/metrics data models. There is a proprietary data model used by some business units. This model support enumerations in value, that is while filling up value for a key, developers can select from a enum. For example log_category key value could be from a list of "Service", "Debug", "Performance" etc. This is to enforce semantics across business units. Is it possible to have similar semantics in logs/metrics data models of OpenTelemetry, either now or planned for future?   Does the SDK support it?

Thanks,
Deep

**Comment by tigrannajaryan** [comment]
There is no enum data type in metric or log data models per se, however you can decide on the list of valid values for the enum and record the particular value as a string attribute. If efficiency is a concern you can also use integer indexes instead of string values.

**Reply by deepkumar-git** [reply]
Thanks for quick response @tigrannajaryan. What your answer clarify is that, it's up to the client to limit any possible values for semantics

**Reply by tigrannajaryan** [reply]
Yes, that's correct.
